Flaskblog/routes.py

```diff
@@ -118,4 +119,31 @@ def new_post():
         flash('Your post has been created succeccfuly.', 'success')
         return redirect(url_for('home'))
 
-    return render_template('new_post.html', title='New Post', form=form)
+    return render_template('new_post.html', title='New Post', legend='New Post', form=form)
+
+
+@app.route('/post/<int:post_id>')
+def post(post_id):
+    post = Post.query.get_or_404(post_id)
+    return render_template('post.html', title=f'Post {post_id}', post=post)
+
+
+@app.route('/post/<int:post_id>/update', methods=['GET', 'POST'])
+@login_required
+def update_post(post_id):
+    post = Post.query.get_or_404(post_id)
+    if post.author != current_user:
+        abort(403)
+
```

Comment:
NOTE: The user in the current session is responsible for making updates on their posts only.